src/jsp/jsp_cl.cpp

```diff
@@ -1219,28 +1288,45 @@ jsp_make_method_sig_list (PARSER_CONTEXT *parser, PT_NODE *node, method_sig_list
 	    sig->method_arg_pos[i] = i;
 	  }
 
+	sig->class_name = nullptr;
+
 
-	sig->arg_info.arg_mode = (int *) db_private_alloc (NULL, (sig_num_args + 1) * sizeof (int));
-	if (!sig->arg_info.arg_mode)
+	sig->arg_info = (METHOD_ARG_INFO *) db_private_alloc (NULL, sizeof (METHOD_ARG_INFO));
+	if (!sig->arg_info)
 	  {
 	    error = ER_OUT_OF_VIRTUAL_MEMORY;
 	    goto end;
 	  }
 
-	sig->arg_info.arg_type = (int *) db_private_alloc (NULL, (sig_num_args + 1) * sizeof (int));
-	if (!sig->arg_info.arg_type)
+	if (sig_num_args > 0)
 	  {
-	    error = ER_OUT_OF_VIRTUAL_MEMORY;
-	    goto end;
-	  }
+	    sig->arg_info->arg_mode = (int *) db_private_alloc (NULL, (sig_num_args) * sizeof (int));
+	    if (!sig->arg_info->arg_mode)
+	      {
+		error = ER_OUT_OF_VIRTUAL_MEMORY;
+		goto end;
```

Comment:
When you get to this, sig->arg_info->arg_type may be a non-nullptr garbage value.

```suggestion
		error = ER_OUT_OF_VIRTUAL_MEMORY;
		sig->arg_info->arg_type = nullptr;
		goto end;
```

Reply:
fixed